File: sylee723/5주차/PGS_모음사전.java
class Solution {
    static int count, answer;
    public int solution(String word) {
        count = 0;
        answer = 0;
        perm("", word);
        
        return answer;
    }
    
    static void perm(String str, String word) {
        if (str.equals(word)) {
            answer = count;
            return;
        }
        count++;
        
        if (str.length() == 5) {
            return;
        }
        perm(str+"A", word);
        perm(str+"E", word);
        perm(str+"I", word);
        perm(str+"O", word);
        perm(str+"U", word);
    }
}

File: daehwa_park/8주차/PGS_모음사전.py
answer = 0
is_same = False

def solution(word): 
    perm(0, "", word)
    
    return answer

def perm(cnt, cur, word):
    global answer, is_same
    mo = ['A', 'E', 'I', 'O', 'U']
    
    if cnt == 5 or is_same:
        return
    
    for m in mo:
        if cur + m == word:
            is_same = True
            answer += 1
            return
        
        if not is_same:
            answer += 1
            perm(cnt + 1, cur + m, word)